docs/hub/agents.md

```diff
@@ -0,0 +1,97 @@
+# Agents on the Hub
+
+This page compiles all the libraries and tools Hugging Face offers for agentic workflows.
+
+## smolagents
+
+[smolagents](https://github.com/huggingface/smolagents) is a lightweight library to cover all agentic use cases, from code-writing agents to computer use, in few lines of code. It is model agnostic, supporting local models served with Hugging Face Transformers, as well as models offered with [Inference Providers](../inference-providers/index.md), and proprietary model providers. 
+
+It offers three agent classes based on the ReAct framework: `CodeAgent` for agents writing their own code, `ToolCallingAgent` for tool calling agents, and the `MultiStepAgent` (on which the two former agents are based) for multi-step ReAct workflows.
+
+If you want to avoid defining agents yourself, the easiest way to start an agent is through the CLI, using the `smolagent` command.
+
+```python
+smolagent "Plan a trip to Tokyo, Kyoto and Osaka between Mar 28 and Apr 7."  --model-type "InferenceClientModel" --model-id "Qwen/Qwen2.5-Coder-32B-Instruct" --imports "pandas numpy" --tools "web_search"
```

Comment:
can we format it on multiple lines? see current rendering: https://moon-ci-docs.huggingface.co/docs/hub/pr_1728/en/agents

Reply:
made a suggestion on above here: https://github.com/huggingface/hub-docs/pull/1728#discussion_r2081098046